src/Files.App/Data/Models/ItemViewModel.cs

```diff
@@ -1064,6 +1065,14 @@ public async Task LoadExtendedItemPropertiesAsync(ListedItem item, uint thumbnai
 
 									SetFileTag(item);
 									wasSyncStatusLoaded = true;
+
+									var cancellationTokenSource = new CancellationTokenSource(3000);
+									// Loop until cached thumbnail is loaded or timeout is reached
+									while (!await LoadItemThumbnailAsync(item, thumbnailSize, matchingStorageFile))
+									{
+										await Task.Delay(100);
+										cancellationTokenSource.Token.ThrowIfCancellationRequested();
```

Comment:
```suggestion
										cancellationTokenSource.Token.ThrowIfCancellationRequested();
										await Task.Delay(100);
```
Better to check the timeout first.